File: 4_Klassen/Objective-C/Bucketlist/Bucketlist/List.m
//
//  List.m
//  Bucketlist
//
//  Created by Jan Brinkmann on 28/06/16.
//  Copyright © 2016 Jan Brinkmann. All rights reserved.
//

#import "List.h"

@interface List ()
@property (nonatomic) NSMutableSet *items;
@end


@implementation List

- (id) init {
    self = [super init];
    
    _items = [[NSMutableSet alloc] init];

    return self;
}

- (void)addItem:(NSString *)newItem
{
    [[self items] addObject:newItem];
}

- (void)removeItem:(NSString *)item
{
    [[self items] removeObject:item];
}

- (void)printList
{
    for (NSString *item in [self items]) {
        printf("%s\n", [item UTF8String]);
    }
}

- (BOOL)isOnList:(NSString *)item
{
    if ([[self items] containsObject:item]) {
        return true;
    }
    
    return NO;
}

@end

File: 4_Klassen/Swift/Bucketlist/Bucketlist/List.swift
//
//  List.swift
//  Bucketlist
//
//  Created by Jan Brinkmann on 28/06/16.
//  Copyright © 2016 Jan Brinkmann. All rights reserved.
//

import Foundation

class List {
    var items = Set<String>()
    
    func addItem(newItem: String) {
        items.insert(newItem)
    }
    
    func resolveItem(item: String) {
        items.remove(item)
    }
    
    func isOnList(item: String) -> Bool {
        return items.contains(item)
    }
    
    func printList() {
        for item in items {
            print(item)
        }
    }
}